File: latex-to-leaflet/examples/convert.rs
//! Convert a LaTeX file to Leaflet JSON.
//!
//! Usage:
//!     cargo run -p latex-to-leaflet --example convert -- examples/sample.tex

use std::env;
use std::fs;
use std::process;

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() != 2 {
        eprintln!("Usage: {} <file.tex>", args[0]);
        process::exit(1);
    }

    let path = &args[1];
    let source = fs::read(path).unwrap_or_else(|e| {
        eprintln!("Error reading {}: {}", path, e);
        process::exit(1);
    });

    let schema = latex_to_leaflet::parse_latex_to_leaflet(&source, path).unwrap_or_else(|e| {
        eprintln!("Parse error: {}", e);
        process::exit(1);
    });

    let json = leaflet_protocol::emit_leaflet_document(&schema).unwrap_or_else(|e| {
        eprintln!("Emit error: {}", e);
        process::exit(1);
    });

    println!("{}", serde_json::to_string_pretty(&json).unwrap());
}

File: sdk/latex-to-leaflet/examples/convert.mjs
#!/usr/bin/env node
/**
 * Example: Convert a LaTeX fixture to Leaflet JSON.
 *
 * Usage (from sdk/latex-to-leaflet):
 *   node examples/convert.mjs
 */

import { readFileSync, writeFileSync } from 'node:fs';
import { resolve, dirname } from 'node:path';
import { fileURLToPath } from 'node:url';
import { convertLatex } from '../index.js';

const __dirname = dirname(fileURLToPath(import.meta.url));

const FIXTURE = resolve(__dirname, 'fixtures', 'sample.tex');
const OUTPUT = resolve(__dirname, 'fixtures', 'sample.leaflet.json');

async function main() {
  const source = readFileSync(FIXTURE, 'utf-8');
  console.error(`Converting ${FIXTURE} ...`);

  const doc = await convertLatex(source);

  const json = JSON.stringify(doc, null, 2);
  writeFileSync(OUTPUT, json + '\n', 'utf-8');

  console.error(`Wrote Leaflet JSON to ${OUTPUT}`);
  console.log(json);
}

main().catch((err) => {
  console.error('Conversion failed:', err.message);
  process.exit(1);
});
